oxli/__init__.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+#
+# This file is a part of khmer, http://github.com/ged-lab/khmer/, and is
+# Copyright (C) Michigan State University, 2015. It is licensed under the
+# three-clause BSD license; see doc/LICENSE.txt.
+# Contact: khmer-project@idyll.org
+#
+
+"""
+Single entry point script for khmer
+"""
+
+import argparse
+import textwrap
+from khmer import khmer_args
+from oxli import build_graph
+
+
+def get_parser():
+    """
+    returns the parser object for the oxli subcommand handler
+    """
+
+    parser = argparse.ArgumentParser(
+        description='Single entry point script for khmer',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    subparsers = parser.add_subparsers()
+
+    # build-graph (formerly load-graph) parsers here
+    parser_build_graph = \
+        subparsers.add_parser('build-graph',
+                              help="Load sequences into the compressible graph"
+                              "format plus optional tagset",
+                              description="Load sequences into the "
+                              "compressible graph format plus optional tagset")
+
+    khmer_args.build_hashbits_args("Load sequences into the compressible"
+                                   "graph format plus optional tagset.",
+                                   None, parser=parser_build_graph)
+    build_graph.build_parser(parser_build_graph)
+    parser_build_graph.set_defaults(func=execute_build_graph)
+
+    return parser
+
+
+def execute_build_graph(args):
+    """
+    Runs the main() in build_graph
+    """
+    build_graph.main(args)
+
+
+def main():
+    """
+    main function; does the parsing and kicks off the subcommand
+    """
+    args = get_parser().parse_args()
+    args.func(args)
+    return 0
```

Comment:
I suggested this, because the way we run scripts for tests. Full discussion on the `ged-lab/dev-flood` gitter channel, but:

What I think it’s happening here is that we don’t have a ‘pure’ shell running the script, so when we run from the shell the None returned is converted to 0
but in our runscript() it’s actually inside Python
and the return code just stays None